src/functional/ffront/type_info.py

```diff
@@ -89,18 +89,24 @@ def function_signature_incompatibilities_fieldop(
 def function_signature_incompatibilities_scanop(
     scanop_type: ts.ScanOperatorType, args: list[ts.TypeSpec], kwargs: dict[str, ts.TypeSpec]
 ) -> Iterator[str]:
-    if not all(is_field_type_or_tuple_of_field_type(arg) for arg in args):
-        yield "Arguments to scan operator must be fields or tuples thereof."
+    if not all(
+        is_field_type_or_tuple_of_field_type(arg) or isinstance(arg, ts.ScalarType) for arg in args
+    ):
+        yield "Arguments to scan operator must be fields, scalars or tuples thereof."
         return
 
-    arg_dims = [extract_dims(el) for arg in args for el in primitive_constituents(arg)]
+    new_args = [
+        ts.FieldType(dims=[], dtype=extract_dtype(arg)) if is_number(arg) else arg for arg in args
```

Comment:
```suggestion
        ts.FieldType(dims=[], dtype=extract_dtype(arg)) if is_instance(arg, ts.ScalarType) else arg for arg in args
```
to be consistent with the check above?